### DISCUSSION TITLE
Running CapRover on the local network

### DISCUSSION THREAD

**Discussion by obizbu** [caprover/caprover#1016] [Q&A]
I followed the instructions on https://caprover.com/docs/run-locally.html on a machine (server) with a fresh installation of Ubuntu 20.04

Everything went perfectly

On my laptop (client) I can access my server IP address at both ports 80 (which has nothing on it) and 3000 (admin interface)

I was able to successfully update the root domain from the admin interface to captain.localhost
I was able to successfully deploy MonicaHQ from the one-click deploy interface, which I deployed to monk.captain.localhost

On my client machine, I updated the etc/hosts file to include:
`192.168.1.99 monk.captain.localhost captain.localhost`

How do I actually access the application? On the client, when I go to monk.captain.localhost, Chrome tells me "This site can’t be reached" and "ERR_CONNECTION_REFUSED". I tried to go to monk.captain.192.168.1.99/ but that doesn't load anything either...

What else do I need to do?

**Comment by githubsaturn** [comment]
What address do you use to reach your dashboard from your local machine?
Is it `http://captain.captain.localhost` ?

**Reply by obizbu** [reply]
No, I used my server IP 192.168.1.99:3000
http://captain.captain.localhost gives me the same error as above

**Reply by githubsaturn** [reply]
Okay then - something is wrong with your name resolution. Try changing your root domain to `captain.local` (instead of `.localhost`).

Update your `/etc/hosts` and your root domain on the dashboard. Then try again.

**Reply by obizbu** [reply]
Actually, I think that issue might be specific to Chrome. I tried again in Safari, and monk.captain.localhost loads, but it just loads up a blue screen saying "Nothing here yet"

![image](https://user-images.githubusercontent.com/1897314/107969130-5609a580-6f7d-11eb-8bf2-51eb330df350.png)

Edit:
Oh, and in Safari, captain.captain.localhost loads just fine

**Reply by githubsaturn** [reply]
"Nothing is here" displayed if the app name doesn't exist. Are you sure you app name is `monk` ? 

On your server run this:
```
cat /captain/generated/nginx/conf.d/captain.conf | grep server_name
```

Do you see `monk.captain.localhost` there?

**Reply by obizbu** [reply]
So I think I narrowed down the issue. Safari keeps redirecting me, while the app wants to redirect me to monk.captain.localhost/register... so Safari ends up loading https://monk.captain.localhost/register

I was able to access the app page by changing to http, however when I fill out the registration form, the redirect gets overtaken by Safari and again I'm forced to https://monk.captain.localhost/register

I tried the solution on this [link](https://apple.stackexchange.com/questions/227662/how-do-i-fully-flush-cached-redirects-from-safari), but it didn't do anything for me...

Is there a way to force CR to load the site even when accessing via https?

**Reply by githubsaturn** [reply]
It's not CapRover at that point, it's Monica: https://github.com/monicahq/monica/issues/176

**Reply by obizbu** [reply]
You're totally right - I had to chance one of the kv values to dev instead of production and that seems to have solved it. 
Now - I can use caprover, and apparently any installed apps but as long as I use them only in Safari (until I work out how to get Chrome to respect the etc/hosts file)

Thanks Kasra! Again, LOVE this project
